common/types/provider_test.go

```diff
@@ -758,12 +758,19 @@ func TestNativeToValue_Primitive(t *testing.T) {
 	expectNativeToValue(t, &rBytes, rBytes)
 
 	// Extensions to core types.
+	expectNativeToValue(t, testInt(1), Int(1))
```

Comment:
Could you also add tests for `expectValueToNative` to make sure the adaption roundtrips appropriately?

Reply:
Added tests and fixed the code as needed